pkg/image/admission/admission.go

```diff
@@ -0,0 +1,319 @@
+
+var _ = oadmission.WantsOpenshiftClient(&imageLimitRangerPlugin{})
+var _ = oadmission.Validator(&imageLimitRangerPlugin{})
+
+// NewImageLimitRangerPlugin provides a new imageLimitRangerPlugin.  Also see SetOpenshiftClient.
+func NewImageLimitRangerPlugin(client clientset.Interface, config io.Reader) (*imageLimitRangerPlugin, error) {
+	plugin := &imageLimitRangerPlugin{
+		Handler:          kadmission.NewHandler(kadmission.Create, kadmission.Update),
+		imageStreamStore: cache.NewStore(cache.MetaNamespaceKeyFunc),
+	}
+	limitRanger, err := limitranger.NewLimitRanger(client, plugin)
+	if err != nil {
+		return nil, err
+	}
+	plugin.limitRanger = limitRanger
+
+	// NOTE: the reflector is set up in SetOpenshiftClient to ensure we have access to the client
+	return plugin, nil
+}
+
+// Run starts the reflectors used by the plugin.
+func (a *imageLimitRangerPlugin) Run() {
+	if a.stopChan == nil {
+		a.stopChan = make(chan struct{})
+		a.imageStreamReflector.RunUntil(a.stopChan)
+	}
+}
+
+// Stop stops the reflectors used by the plugin.
+func (a *imageLimitRangerPlugin) Stop() {
+	if a.stopChan != nil {
+		close(a.stopChan)
+		a.stopChan = nil
+	}
+}
+
+// Admit invokes the admission logic for checking against LimitRanges.
+func (a *imageLimitRangerPlugin) Admit(attr kadmission.Attributes) error {
+	if !a.SupportsAttributes(attr) {
+		return nil // not applicable
+	}
+
+	// If the namespace is not set on the object ensure it gets set.
+	if om, err := kapi.ObjectMetaFor(attr.GetObject()); err == nil {
+		if om.Namespace == "" {
+			glog.V(5).Infof("%s: namespace was not set for object, defaulting to %s", pluginName, attr.GetNamespace())
+			om.Namespace = attr.GetNamespace()
+		}
+	}
+
+	return a.limitRanger.Admit(attr)
+}
+
+// SupportsAttributes is a helper that returns true if the resource is supported by the plugin.
+// Implements the LimitRangerActions interface.
+func (a *imageLimitRangerPlugin) SupportsAttributes(attr kadmission.Attributes) bool {
+	if attr.GetSubresource() != "" {
+		return false
+	}
+
+	resource := attr.GetResource()
+	return resource == imageapi.Resource("imagestreammappings") ||
+		resource == imageapi.Resource("imagestreamtags") ||
+		resource == imageapi.Resource("imagestreams")
+}
+
+// SupportsLimit provides a check to see if the limitRange is applicable to image objects.
+// Implements the LimitRangerActions interface.
+func (a *imageLimitRangerPlugin) SupportsLimit(limitRange *kapi.LimitRange) bool {
+	if limitRange != nil {
+		for _, limit := range limitRange.Spec.Limits {
+			if limit.Type == imageapi.LimitTypeImageSize {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// Limit is the limit range implementation that checks resource against the
+// image limit ranges.
+// Implements the LimitRangerActions interface
+func (a *imageLimitRangerPlugin) Limit(limitRange *kapi.LimitRange, kind string, obj runtime.Object) error {
+	var images map[string]imageapi.Image
+
+	switch obj.(type) {
```

Comment:
`switch isobj := obj.(type)` will allow you to get rid of the casts you're doing in every case and you'll just use `isobj` as the appropriate type already.